File: graal-nodejs/test/graal/unit/undefined.cc
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

#define SUITE Undefined

// Undefined/MissingPrimitive::IsUndefined

EXPORT_TO_JS(IsUndefined) {
    args.GetReturnValue().Set(Undefined(args.GetIsolate())->IsUndefined());
}

// Undefined/MissingPrimitive::IsNull

EXPORT_TO_JS(IsNull) {
    args.GetReturnValue().Set(Undefined(args.GetIsolate())->IsNull());
}

// Undefined/MissingPrimitive::IsTrue

EXPORT_TO_JS(IsTrue) {
    args.GetReturnValue().Set(Undefined(args.GetIsolate())->IsTrue());
}

// Undefined/MissingPrimitive::IsFalse

EXPORT_TO_JS(IsFalse) {
    args.GetReturnValue().Set(Undefined(args.GetIsolate())->IsFalse());
}

#undef SUITE

File: graal-nodejs/test/graal/unit/undefined.js
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

var assert = require('assert');
var module = require('./_unit');

describe('Undefined', function () {
    describe('IsUndefined', function () {
        it('should return true', function () {
            assert.strictEqual(module.Undefined_IsUndefined(), true);
        });
    });
    describe('IsNull', function () {
        it('should return false', function () {
            assert.strictEqual(module.Undefined_IsNull(), false);
        });
    });
    describe('IsTrue', function () {
        it('should return false', function () {
            assert.strictEqual(module.Undefined_IsTrue(), false);
        });
    });
    describe('IsFalse', function () {
        it('should return false', function () {
            assert.strictEqual(module.Undefined_IsFalse(), false);
        });
    });
    describe('IsNativeError', function () {
        it('should return false', function () {
            assert.strictEqual(module.Value_IsNativeError(undefined), false);
        });
    });
});
